### DISCUSSION TITLE
Using apis.RequestData handling OnRecordAfterAuthWithOAuth2Request has missing body parameters from authWithOAuth2 method

### DISCUSSION THREAD

**Discussion by pnmcosta** [pocketbase/pocketbase#2535] [Q&A]
Hi,

Is it expected that the HttpContext on the `OnRecordAfterAuthWithOAuth2Request` event not to include the `body` parameters sent with the js SDK's `authWithOAuth2` method?

They are there on the `OnRecordBeforeAuthWithOAuth2Request` event as expected but I would like to do some actions after the whole oauth process was successful based on those parameters. I am only seeing what I've passed on `createData`, as if the original request data was replaced somehow.

Thank you,

**Comment by pnmcosta** [comment]
I suppose [this](https://github.com/pocketbase/pocketbase/blob/master/apis/record_auth.go#L194) could be it, I believe it's mutating the cached `apis.RequestData` in the context and I do use it too on my app.

In the meantime to get around this I'm using my own context entry with the wanted parameters.

**Reply by ganigeorgiev** [reply]
> I believe it's mutating the cached apis.RequestData in the context and I do use it too on my app.

Yes, I'll consider changing it sometime tomorrow and will push it as v0.16.1.

**Reply by pnmcosta** [reply]
Thank you, great support 💪
